### PR TITLE
feat: add new table ref_commits for the refdiff plugin

### PR DESCRIPTION
**Pull request by abeizn**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
1. add new table ref_commits for the refdiff plugin.
2. update some struct name

### Does this close any open issues?
related to #3685 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: models/domainlayer/code/commits_diff.go [hunk 1]
```diff
@@ -28,11 +28,22 @@ func (CommitsDiff) TableName() string {
 	return "commits_diffs"
 }
 
-type FinishedCommitsDiffs struct {
+type RefCommit struct {
```
Comment (likyh) [inline models/domainlayer/code/commits_diff.go:31]: Why does this table need to be old and new? It seems that one refId and one commitSha are enough.

File: plugins/refdiff/tasks/commit_diff_calculator.go [hunk 2]
```diff
@@ -41,18 +42,29 @@ func CalculateCommitsDiff(taskCtx core.SubTaskContext) errors.Error {
...
+				newRefId := fmt.Sprintf("%s:%s", repoId, pair[2])
+				oldRefId := fmt.Sprintf("%s:%s", repoId, pair[3])
+				if pair[2] != newRefId || pair[3] != oldRefId {
+					refCommit.NewCommitSha = pair[0]
+					refCommit.OldCommitSha = pair[1]
+					refCommit.NewRefId = newRefId
+					refCommit.OldRefId = oldRefId
+				}
+
 				commitPairs = append(commitPairs[:i], commitPairs[i+1:]...)
```
Comment (mindlesscloud) [inline plugins/refdiff/tasks/commit_diff_calculator.go:68]: Removing items from the `commitPairs` would affect the underlying `data.Options.AllPairs` which is shared among all subtasks.

File: models/domainlayer/code/commits_diff.go (hunk 1 above)
Comment (abeizn) [reply models/domainlayer/code/commits_diff.go:31]: yes, one is not enough

Comment (likyh) [reply models/domainlayer/code/commits_diff.go:31]: Maybe adding column in finished_commit_diffs is better. The solution that create A new table is just move duplicate row to new table.

Comment (abeizn) [reply models/domainlayer/code/commits_diff.go:31]: no, because the deployment does not use it in finished_commit_diffs

Comment (klesh) [reply models/domainlayer/code/commits_diff.go:31]: If one needs to know which `commit-sha` a `ref` is pointing to, one should use the `refs` table.
This table is designed to store the pairs that are filtered by the `refdiff` tag regex.

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**